File: play-2.5.x/K-Modules/kmodules/K-Base/app/models/K/EbeanConfig/MyTransactionEventListener.java
package models.K.EbeanConfig;

import K.Service.NotifyService;
import com.avaje.ebean.Transaction;
import com.avaje.ebean.event.TransactionEventListener;
import models.K.BgTask.PlanTask;
import play.Logger;

/**
 * Created by kk on 15/2/2.
 */
public class MyTransactionEventListener implements TransactionEventListener {

    public MyTransactionEventListener() {
        Logger.debug("Ebean server load MyTransactionEventListener");
    }

    @Override
    public void postTransactionCommit(Transaction tx) {
        if (tx.getUserObject(PlanTask.class.getName()) != null) {
            NotifyService notifyService = NotifyService.getService();
            if (notifyService != null) {
                notifyService.NotifyNewPlanTask();
//                Logger.debug("MyTransactionEventListener::notifyService.NotifyNewPlanTask()");
            }
        }
        Object obj = tx.getUserObject(MyBeanPersistController.class.getName());
        if (obj != null) {
            StringBuilder sb = (StringBuilder) obj;
            Logger.of("PersistLog").info(sb.toString());
        }
    }

    @Override
    public void postTransactionRollback(Transaction tx, Throwable cause) {

    }
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/models/K/EbeanConfig/MyTransactionEventListener.kt
package models.K.EbeanConfig

import K.Common.Helper
import K.Service.NotifyService
import com.avaje.ebean.Transaction
import com.avaje.ebean.event.TransactionEventListener
import models.K.BgTask.PlanTask
import play.Logger

/**
 * Created by kk on 15/2/2.
 */
class MyTransactionEventListener : TransactionEventListener {

    init {
        Helper.DLog("Ebean server load MyTransactionEventListener")
    }

    override fun postTransactionCommit(tx: Transaction) {
        if (tx.getUserObject(PlanTask::class.java.name) != null) {
            val notifyService = NotifyService.getService()
            notifyService?.NotifyNewPlanTask()
        }
        val obj = tx.getUserObject(MyBeanPersistController::class.java.name)
        if (obj != null) {
            val sb = obj as StringBuilder
            Logger.of("PersistLog").info(sb.toString())
        }
    }

    override fun postTransactionRollback(tx: Transaction, cause: Throwable) {

    }
}
